src/App.tsx

```diff
@@ -9,23 +9,31 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
-import { Counter } from './features/counter/Counter';
+import { useAppSelector, useAppDispatch } from './app/hooks';
+import { currentAuthor } from './features/author/authorSlice';
+import {
+  currentPost,
+  allPost,
+  setPosts,
+} from './features/posts/postsSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
+  const author = useAppSelector(currentAuthor);
+  const posts = useAppSelector(allPost);
+  const dispatch = useAppDispatch();
+  const selectedPost = useAppSelector(currentPost);
   const [loaded, setLoaded] = useState(false);
   const [hasError, setError] = useState(false);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  // const [selectedPost, setSelectedPost] = useState<Post | null>(null);
 
   function loadUserPosts(userId: number) {
     setLoaded(false);
 
     getUserPosts(userId)
-      .then(setPosts)
+      .then(res => {
+        dispatch(setPosts(res));
+      })
```

Comment:
You should use `createAsyncThunk` from the videos and docs. Please, read more about it and use it everywhere you make queries.